mtags/src/main/scala-3/scala/meta/internal/pc/completions/Completions.scala

```diff
@@ -50,15 +52,84 @@ class Completions(
           .flatMap(CompletionValue.fromCompiler)
           .filterInteresting()
 
-    val args = NamedArgCompletions.contribute(pos, path)
-    val keywords = KeywordsCompletions.contribute(path, completionPos)
-    val all = completions ++ args ++ keywords
+    val (advanced, exclusive) = advancedCompletions(path, pos, completionPos)
+    val all =
+      if exclusive then advanced else completions ++ advanced
```

Comment:
I would adjust this part a bit to not to call `Completion.completions` if there is no need in them.

```scala
def completions(): (List[CompletionValue], SymbolSearch.Result) = 
  val (advanced, exclusive) = advancedCompletions(path, pos, completionPos)
  val (all, result) = 
     if exclusive then
        (advanced, SymbolSearch.Result.COMPLETE)
     else
        path match 
           case Nil if pos.source.file.extension != "sc" =>
               (List.empty, SymbolSearch.Result.COMPLETE)
            case _ =>
                val (_, compilerCompletions) = Completion.completions(pos)
                val (compiler, result) = compilerCompletions
                     .flatMap(CompletionValue.fromCompiler)
                     .filterInteresting()
                  (advanced ++ compiler, result)
    ...
                   
```

Reply:
Thank you for the suggestion! done https://github.com/scalameta/metals/pull/3897/commits/c047096799d2f497352ea28574b68aa53b7f48a8